### PR TITLE
Initial mesh coloring

### PR DESCRIPTION
**Pull request by aerkiaga**
This transfers the computed partial charges to the mesh. As it stands, there are some areas to polish:

* Selecting `None` doesn't allow mesh colors to be set via the `Meshes` plugin.
* ~The colormap can't be changed yet.~
* The charge model can't be set yet.

Signed-off-by: Aritz Erkiaga <aerkiaga3@gmail.com>

Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

### CODE REVIEW

**Comment by aerkiaga** [comment]
image

**Review by ghutchis — CHANGES_REQUESTED** [review]
Just a few minor changes

File: avogadro/core/mesh.cpp [hunk 1]
```diff
@@ -118,6 +118,14 @@ const Core::Array<Color3f>& Mesh::colors() const
   return m_colors;
 }
 
+Core::Array<Vector3ub>* Mesh::colorsRGB() const
```
[the remark this answers is not in the document]
Comment (aerkiaga) [reply avogadro/core/mesh.cpp:121]: `addVertices` needs RGB, but I agree that this is far from an optimal solution...

**Comment by aerkiaga** [comment]
I have chosen to make the loop itself concurrent rather than using `potentials()` as the latter seems to just run Open Babel many many times then lock up... I've not managed to find the cause of this behavior.

**Comment by ghutchis** [comment]
> I have chosen to make the loop itself concurrent rather than using potentials() as the latter seems to just run Open Babel many many times then lock up

Please file a bug. I'll take a look - it should just get the charges, cache them and be done with `obabel`

**Pull request merged**